src/Tizen.System/PowerUsage/PowerUsage.cs

```diff
@@ -0,0 +1,224 @@
+﻿/*
+ * Copyright (c) 2020 Samsung Electronics Co., Ltd All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the License);
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an AS IS BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+using System;
+using System.Collections.Generic;
+using System.Runtime.InteropServices;
+
+namespace Tizen.System
+{
+    static internal class Globals
+    {
+        internal const string LogTag = "Tizen.System.PowerUsage";
+    }
+
+    /// <summary>
+    /// Enumeration for battery consuming features.
+    /// </summary>
+    /// <since_tizen> 7 </since_tizen>
+    public enum ResourceType
+    {
+        /// <summary>
+        /// Bluetooth Low Energy.
+        /// </summary>
+        Ble = 0,
+
+        /// <summary>
+        /// Wi-Fi.
+        /// </summary>
+        Wifi = 1,
+
+        /// <summary>
+        /// CPU.
+        /// </summary>
+        Cpu = 2,
+
+        /// <summary>
+        /// Display.
+        /// </summary>
+        Display = 3,
+
+        /// <summary>
+        /// Device Network.
+        /// </summary>
+        DeviceNetwork = 4,
+
+        /// <summary>
+        /// GPS Sensor.
+        /// </summary>
+        Gps = 5
+    }
+
+    /// <summary>
+    /// Provides information related to the power consumption by applications or by hardware resources on a battery-powered device for a given duration of time.
+    /// </summary>
+    /// <feature>http://tizen.org/feature/battery</feature>
+    /// <privilege>http://tizen.org/privilege/systemmonitor</privilege>
+    /// <since_tizen> 7 </since_tizen>
+    public static class PowerUsage
+    {
+        static PowerUsage()
+        {
+        }
+
+        private static double GetPowerUsage(IntPtr BMHandle, ResourceType rtype)
+        {
+            double batteryUsage = 0;
+            int ret = Interop.PowerUsage.GetPowerUsage(BMHandle, rtype, out batteryUsage);
+            if (((PowerUsageError)ret != PowerUsageError.None))
+            {
+                Log.Error(Globals.LogTag, "Error getting battery usage for resource" + (PowerUsageError)ret);
+                throw PowerUsageErrorFactory.ThrowPowerUsageException(ret);
+            }
+            return batteryUsage;
+        }
+
+        private static long ConvertDateTimeToTimestamp(DateTime dateTime)
```

Comment:
How about to use it
https://docs.microsoft.com/en-us/dotnet/api/system.datetimeoffset.tounixtimeseconds?redirectedfrom=MSDN&view=netframework-4.8#System_DateTimeOffset_ToUnixTimeSeconds